features/graph/simple/src/main/java/org/opennms/netmgt/graph/simple/SimpleEdge.java

```diff
@@ -29,37 +29,73 @@
 // TODO MVR fix package name opennsm vs opennms
 package org.opennms.netmgt.graph.simple;
 
-import org.opennms.netmgt.graph.api.AbstractEdge;
+import java.util.Objects;
+
+import org.opennms.netmgt.graph.api.Edge;
 import org.opennms.netmgt.graph.api.VertexRef;
 import org.opennms.netmgt.graph.api.generic.GenericEdge;
-import org.opennms.netmgt.graph.api.generic.GenericProperties;
 
-public class SimpleEdge extends AbstractEdge<SimpleVertex> {
+/**
+ * Acts as a domain specific view on a Edge.
+ * Can be extended by domain specific edge classes.
+ * It contains no data of it's own but operates on the data of it's wrapped GenericEdge.
+ */
+public class SimpleEdge implements Edge {
+
+    protected GenericEdge delegate;
 
     public SimpleEdge(SimpleVertex source, SimpleVertex target) {
-        super(source, target);
+        delegate = new GenericEdge(source.asGenericVertex(), target.asGenericVertex());
+    }
+
+    public SimpleEdge(GenericEdge genericEdge) {
+        this.delegate = genericEdge;
     }
 
     public SimpleEdge(SimpleEdge copyMe) {
+        // copy the delegate to have a clone down to the properties maps
+        this(new GenericEdge(copyMe.asGenericEdge()));
+
+        // TODO: patrick: discuss with mvr: is this the expected behaviour?
```

Comment:
Currently we only allow references to the same namespace, e.g. `new SimpleEdge(simpleVertex1, simpleVertex2)`. However we need to support references to another namespace (even in another graph container) as well. To support this, the edge should not contain the vertex, but only the reference.

I am at the moment not sure if we can use the "delegate pattern" then here.
If the tests are still passing, I'd opt for just leaving it as is for now, as this must be reworked when supporting references to other namespaces.

Reply:
ok, I'll leave it for now and make a TODO for us.